### DISCUSSION TITLE
How to make a internet using python.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#133463] [Programming Help]
### Body

So, I love the thought of internet, and how it works, but I recently found a module called pywifi. Though it doesn't work on Mac. Do you know any alternatives? And I also like to make my own version of the modules I find interesting, but I have no idea of how to make this.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ghost** [comment]
So... Iv'e encountered a problem. On    def connect(self, ssid, password):
        if self.os == 'Darwin':
            return self.connect_mac(ssid, password)
        else:
            raise NotImplementedError("Macos is only supported")
It raises AttributError: 'str' object has no attribute 'os'. It needs to be Darwin to work, which is a string. Can you help me?

**Reply by Bernardo-Zamin** [reply]
Have you tried searching the stack overflow to see if anyone has ever had the same problem? Try to use some AI to help you

**Comment by CleverLemming1337** [comment]
Where do you use the function and self?

**Comment by vmars316** [comment]
Sorry for offTopic
but i am new here and I don't see any **Post** buttons anywhere .
How to **Post** ?
Thanks for your Help...

**Reply by CleverLemming1337** [reply]
What do you mean with post?

**Reply by mmdoff1** [reply]
There is should be a "+" button in a blue circle at the top right of the screen. (Mobile) I don't know where you are using GitHub tho.

**Comment by vmars316** [comment]
How to 'Post New Thread' ,
Thanks

**Reply by CleverLemming1337** [reply]
If you mean to start a new discussion, you can do so [here](https://github.com/orgs/community/discussions).

(Use the plus button in the app)

**Comment by xsxo** [comment]
start learn socket library
